src/rai_bench/rai_bench/tool_calling_agent/mocked_tools.py

```
@@ -265,45 +268,163 @@ def _run(self, msg_type: str) -> str:
             return str(
                 {
                     "payload": response.payload,
                     "metadata": response.metadata,
                 }
             )
-        else:
-            raise KeyError(
-                f"Model for service type {service_type} not included in models"
-            )
+        except ValueError as e:
+            raise ValueError(f"Failed to populate fields: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)